app/screens/home/recent_mentions/index.tsx

```diff
@@ -27,22 +113,93 @@ const RecentMentionsScreen = () => {
-                style={[{flex: 1, justifyContent: 'center', alignItems: 'center'}, animated]}
+        <>
+            <NavigationHeader
+                isLargeTitle={isLargeTitle}
+                showBackButton={false}
+                subtitle={subtitle}
+                title={title}
+                hasSearch={false}
+                scrollValue={scrollValue}
+                forwardedRef={scrollRef}
+            />
+            <SafeAreaView
+                style={{flex: 1}}
+                edges={['bottom', 'left', 'right']}
             >
-                <Text style={{fontSize: 20, color: '#fff'}}>{'Recent Mentions Screen'}</Text>
-            </Animated.View>
-        </SafeAreaView>
+                <Animated.View style={[{flex: 1}, animated]}>
+                    {loading && !mentions.length ? (
+                        <View style={styles.loading}>
+                            <ActivityIndicator
+                                color={theme.sidebarBg}
+                                size='large'
+                            />
+                        </View>
+                    ) : (
+                        <AnimatedSectionList
+                            ref={scrollRef}
+                            contentContainerStyle={paddingTop}
+                            sections={sections}
+                            scrollToOverflowEnabled={true}
+                            showsVerticalScrollIndicator={false}
+                            progressViewOffset={LARGE_HEADER_PROGRESSVIEW_OFFSET}
+                            scrollEventThrottle={16}
+                            indicatorStyle='black'
+                            onScroll={onScroll}
+                            onRefresh={handleRefresh}
+                            refreshing={refreshing}
+                            renderItem={renderItem}
+                            renderSectionHeader={({section: {title: date}}: {section: Section}) => (
+                                <DateSeparator
+                                    date={new Date(date)}
+                                    theme={theme}
+                                />
+                            )}
+                        />
+                    )}
+                </Animated.View>
+            </SafeAreaView>
+        </>
     );
 };
 
-export default RecentMentionsScreen;
+const enhance = withObservables([], ({database}: WithDatabaseArgs) => ({
+    mentions: database.get<SystemModel>(SYSTEM).query(
+        Q.where('id', SYSTEM_IDENTIFIERS.RECENT_MENTIONS),
+        Q.take(1),
+    ).observeWithColumns(['value']).pipe(
+        switchMap((rows) => {
+            if (!rows.length || !rows[0].value.length) {
+                return of$([]);
+            }
+            const row = rows[0];
+            return database.get(POST).query(
+                Q.where('id', Q.oneOf(row.value)),
+                Q.sortBy('create_at', Q.desc),
+            ).observeWithColumns(['message']);
```

Comment:
why do you need to observe on the message?, isn't it best if the `Mention` component observes the post, that way only that component will re-render instead of re-rendering the entire list

Reply:
done